build/build/src/rust/enso_linter.rs

```diff
@@ -0,0 +1,36 @@
+use super::*;
+
+use ide_ci::programs::cargo;
+use ide_ci::programs::Cargo;
+
+use crate::paths::generated::RepoRoot;
+
+const LINTER_CRATE_NAME: &str = "enso-parser-debug";
+const LINTER_BIN_NAME: &str = "check_syntax";
+const ENSO_EXTENSION: &str = ".enso";
+
+pub fn cargo_run_linter_cmd(repo_root: &Path) -> Result<Command> {
+    let mut ret = Cargo.cmd()?;
```

Comment:
While having such a linter may satisfy our current development needs, I believe the right linter shall be integrated into the `enso` command.

There is `enso --compile` that is being invoked as part of CI already. We don't need a new CI task (in my opinion), but rather we should make sure that `enso --compile` (maybe with `--lint`) fails the CI when the formatting isn't correct.

Reply:
It's important for the compiler to be able to emit syntax warnings. I think it's also useful to have a simple standalone linter--if linting fails in CI, it's nice if it happens right away, not after building the engine. It isn't a new CI task; it's part of `./run lint`.

Reply:
> it's also useful to have a simple standalone linter

Indeed. However such a linter has to be **available to Enso users**, not just to us. E.g. it cannot be part of `./run lint`. It has to be part of `enso-engine*/bin/enso`.

How will we expose the linter to Enso end users? What needs to be changed to make that happen? Can you report an issue describing what needs to be done? Can you implement it (and remove current `./run lint` temporary solution) or shall JVM guys in engine team do it?